src/components/SearchBarPopup/SearchBarPopup.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { RefObject } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import "./searchBarPopup.scss";
+import _ from "lodash";
+
+interface Destination {id: number, name: string, address: string, isAdded: boolean}
+
+/**
+ * Search bar states
+ * searchKey: Keyword to perform search with
+ * results: Search results retrieved from backend call
+ * loading: True if a search is being performed, false if no search currently performed
+ * errMsg: Any error message from search
+ */
+export interface SearchBarState {
+    searchKey: string;
+    results: Array<Destination>;
+    resultsOpen: boolean;
+    loading: boolean;
+    errMsg: string;
+}
+
+/**
+ * A popup to search for places and to be added into a Trecipe
+ */
+export class SearchBarPopup extends React.Component<{}, SearchBarState> {
+    private container: RefObject<HTMLDivElement> = React.createRef<
+        HTMLDivElement
+    >();
+
+    state = {
+        searchKey: "",
+        resultsOpen: false,
+        results: [{ id: 1, name: "Place 1", address: "City, County", isAdded: false },
+            { id: 2, name: "Place 2", address: "City, County", isAdded: false },
+            { id: 3, name: "Place 3", address: "City, County", isAdded: false }],
+        loading: false,
+        errMsg: "",
+    };
+
+    private handleOnSearchInputChange(
+        e: React.ChangeEvent<HTMLInputElement>
+    ): void {
+        const searchKey: string = e.target.value;
+        if (_.isEmpty(searchKey)) {
+            this.setState({ searchKey: searchKey, results: [], errMsg: "" });
+        } else {
+            this.setState({ searchKey: searchKey, loading: true, errMsg: "" }, () =>
+                this.fetchSearchResults(searchKey)
+            );
+        }
+    }
+
+    private fetchSearchResults(
+        searchInput: string,
+        ): void {
+        new Promise<Array<Destination>>(function (resolve, reject) {
+            /** TODO: Modify logic here to make HTTP call to backend for fetching search results.
+             *  Update DB if the previous result list has been changed.
+             */
+
+            let results: Destination[] = [{id: 1, name: "Place 1", address:"City, County", isAdded: false}, 
+                {id: 2, name: "Place 2", address: "City, County", isAdded: false }, 
+                {id: 3, name: "Place 3", address: "City, County", isAdded: false }];
+            resolve(results);
+        })
+            .then((results: Destination[]) => {
+                this.setState({ resultsOpen: true, loading: false, searchKey: searchInput });
+
+                // Todo: this needs to be restored afterwards! 
```

Comment:
Why can't we assign results to state right now?? Realized that after we clear search keyword, the next time we enter something, the result box doesn't show up anymore. Because `results` field in state is never updated after being set to empty.